core/src/com/etheller/warsmash/viewer5/handlers/w3x/simulation/behaviors/build/CBehaviorHumanRepair.java

```diff
@@ -42,17 +44,66 @@ public boolean isWithinRange(CSimulation simulation) {
 
     @Override
     protected CBehavior update(CSimulation simulation, boolean withinFacingWindow) {
-        unit.getUnitAnimationListener().playAnimation(false, AnimationTokens.PrimaryTag.STAND,
-                SequenceUtils.WORK, 1.0f, true);
         if(this.target instanceof CWidget) {
             CWidget targetWidget = (CWidget) this.target;
-            float newLifeValue = targetWidget.getLife() + 1;
-            boolean done = newLifeValue > targetWidget.getMaxLife();
-            if(done) {
-                newLifeValue = targetWidget.getMaxLife();
-            }
-            targetWidget.setLife(simulation, newLifeValue);
-            if(done) {
+            //progress construction here (MFROMAZ)
+            if(targetWidget.getClass()==CUnit.class) {
```

Comment:
So far in Warsmash I have preferred to use `targetWidget instanceof CUnit` instead of `targetWidget.getClass() == CUnit.class`, and there is actually an important difference in a future codebase even though the current code has no difference. Basically, if someone decided to do the (rather silly) idea of subclassing CUnit to make CUnitWard or something, then `(new CUnitWard(...)) instanceof CUnit` would return true but `(new CUnitWard(...).getClass() == CUnit.class` would return false.

This is another fairly pedantic thing and not a substantial issue. If we wanted to be super pedantic about code design, in theory I think it's better to never do either of these and instead use the ability target visitor that I set up, i.e. `target.visit(AbilityTargetVisitor<Boolean>() {...})` and implement some AbilityTargetVisitor who returns true for units. The `instanceof` case, although theoretically better than class equality, can be stupid if we copy CUnit to make some other type like maybe CItem if we did not already have that. The visitor would provide a compile error forcing the human to reconsider what to do with a CItem in this case where it previously was not considered, while the `instanceof` would silently ignore the CItem as a non-unit.

Just a brainstorm on this one. Again, maybe not super important. For the sake of brevity/laziness, as I'm sure you've seen already, I contradicted this knowledge and used `instanceof CUnit` a lot already in Warsmash. The probability of adding additional subclasses of CWidget is extraordinarily low in this project.